lib/tasks/precompute_gene_ontology.rake

```diff
@@ -0,0 +1,209 @@
+MASTER_OWL_URI = "https://raw.githubusercontent.com/arpcard/aro/master/aro.owl".freeze
+URL_PUBMED = "https://www.ncbi.nlm.nih.gov/pubmed/".freeze
+S3_JSON_BUCKET = "idseq-database".freeze
+S3_ARGANNOT_FASTA = "s3://idseq-database/amr/ARGannot_r2.fasta".freeze
+
+DRUG_CLASSES = {
+  "agly" => ["Aminoglycosides", "0000016"],
+  "bla" => ["Beta-lactamases", "3000001"],
+  "colistin" => ["Colistins", "0000067"],
+  # "fcd" => [], # Unclear which one this refers to
+  "fos" => ["Fosfomycin", "0000025"],
+  "fcyn" => ["Fosfomycin", "0000025"],
+  "flq" => ["Fluroquinolones", "0000001"],
+  "gly" => ["Glycopeptides", "3000081"],
+  "mls" => ["Macrolide-lincosamide-streptogramin", "0000000"],
+  "ntmdz" => ["Nitroimidazoles", "3004115"],
+  "oxzln" => ["Oxazolidinones", "3000079"],
+  "phe" => ["Phenicols", "3000387"],
+  "rif" => ["Rifampicin", "3000169"],
+  "sul" => ["Sulfonamides", "3000282"],
+  "tet" => ["Tetracyclines", "3000050"],
+  "tmt" => ["Trimethoprim", "3000188"],
+}.freeze # label, ARO Accession number
+
+require "aws-sdk-s3"
+require "nokogiri"
+
+desc 'Updates and precomputes AMR Ontology information'
+task precompute_gene_ontology: :environment do
+  s3 = Aws::S3::Client.new(region: DEFAULT_S3_REGION)
+
+  owl_uri = URI(MASTER_OWL_URI)
+  response = Net::HTTP.get_response(owl_uri)
+  unless response.is_a?(Net::HTTPSuccess)
+    Rails.logger.error("Unable to fetch OWL file")
+    return
+  end
+  owl_doc = Nokogiri::XML(response.body.force_encoding("utf-8"))
+
+  arg_annot_fasta = nil
+  split_uri = S3_ARGANNOT_FASTA.split("/", 4)
```

Comment:
it looks like you could dry it out by using S3_JSON_BUCKET and just key `amr/ARGannot_r2.fasta`